### PR TITLE
Add methods for verifying presentations.

### PR DESCRIPTION
**Pull request by mikelodder7**

### CODE REVIEW

**Comment by jared-s** [comment]
@mikelodder7 i think a rebase will resolve the CI failures here and in the other PR

File: implementations/rust/ockam/ockam_credential/src/holder.rs [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+                i = j;
+                break;
+            }
+        }
+
+        let pk = issuer_pk
+            .to_public_key(offer.schema.attributes.len())
+            .map_err(|_| CredentialError::InvalidCredentialSchema)?;
+        let mut messages = BTreeMap::new();
+        messages.insert(i, self.id.clone());
```
Comment (jared-s) [inline implementations/rust/ockam/ockam_credential/src/holder.rs:43]: If there are no schema attributes which have a label of `SECRET_ID` then `i` will be the default initialization of 0. Is that acceptable here?

**Review by jared-s — APPROVED** [review]
One question about a conditional. Looks good, could use some test coverage.

**Comment by jared-s** [comment]
@mikelodder7 I'll merge this one next as soon as rebase is resolved

**Review by mrinalwadhwa — APPROVED** [review]

**Comment by jared-s** [comment]
Verified that these commits are on `develop` now. GH messing up again on merged state.

**Pull request merged**